src/System.Text.Json/tests/Utf8JsonReaderTests.cs

```diff
@@ -1796,6 +1796,47 @@ public static void TestJsonReaderUtf8SpecialString(string jsonString, string exp
             }
         }
 
+        [Fact]
+        public static void TestBOMWithSingleJsonValue()
+        {
+            Assert.ThrowsAny<JsonException>(() =>
+            {
+                ReadOnlySpan<byte> Utf8BomAndValue = new byte[] { 0xEF, 0xBB, 0xBF, (byte)'1' };
+                var json = new Utf8JsonReader(Utf8BomAndValue, true, default);
```

Comment:
nit: Rather than duplicating the code, make this test a `[Theory]` and add isFinalbock true/false as `[InlineData]`